api/src/test/java/org/openmrs/test/BaseContextSensitiveTest.java

```diff
@@ -864,7 +859,7 @@ protected IDatabaseConnection setupDatabaseConnection(Connection connection) thr
 	 * 
 	 * @throws Exception
 	 */
-	public void deleteAllData() {
+	public synchronized void deleteAllData() {
```

Comment:
synchronized was added for JUnit 5 tests so adding here as well. It does make sense to synchronize.